[DOC] Create Kafka Bridge quick start

laidan6000 opened the pull request:
Created a Kafka Bridge quickstart for AMQ Streams on RHEL.

Describes how to send requests from local machine.

ppatierno commented on books/proc-bridge-retrieving-latest-messages-from-consumer.adoc:
```diff
@@ -0,0 +1,59 @@
+  -H 'accept: application/vnd.kafka.json.v2+json'
+----
++
+After creating and subscribing to a Kafka Bridge consumer, a first GET request will return an empty response because the poll operation starts a rebalancing process to assign partitions.
```
I would say "triggers" instead of "starts". It's not the poll per-se starting the rebalancing.

laidan6000 replied:
Sure. I'll need to change this in in the Strimzi doc as well.

ppatierno commented on books/proc-bridge-subscribing-consumer-topics.adoc:
```diff
@@ -0,0 +1,35 @@
++
+The `topics` array can contain a single topic (as shown above) or multiple topics. If you want to subscribe the consumer to multiple topics that match a regular expression, you can use the `topic_pattern` string instead of the `topics` array. 
++
+If the request is successful, the Kafka Bridge returns a `204` (No Content) code only.
```
you specified "(No Content)" here and I like the idea. Can you do the same in all the other places where just the status code "204" is mentioned? I would do the same for any status code, always providing the related description.

laidan6000 replied:
OK, that's done. I'll need to update the Strimzi doc as well.

PaulRMellor commented on books/con-loggers-kafka-bridge.adoc:
```diff
@@ -4,7 +4,7 @@
-= Configuring loggers for the {ProductName} Kafka Bridge
+= Configuring loggers for the Kafka Bridge
 
 The {ProductName} Kafka bridge allows you to set a different log level for each operation that is defined by the related OpenAPI specification.
```
Do we want to keep {ProductName} here?

PaulRMellor commented on books/con-requests-kafka-bridge.adoc:
```diff
@@ -25,14 +25,14 @@ Specify data formats and HTTP headers to ensure valid requests are submitted to
 API request and response bodies are always encoded as JSON.
 
-* When performing consumer operations, `POST` requests must provide the following `Content-Type` header:
+* When performing consumer operations, `POST` requests must provide the following `Content-Type` header if there is a non-empty body:
```
'non-empty body' sounds a bit odd. If the body of a request or response is not empty? Should we point out that Content-Type shouldn't be set if a body is empty?

PaulRMellor commented on books/con-requests-kafka-bridge.adoc:
```diff
@@ -48,13 +48,15 @@ m¦Content-Type: application/vnd.kafka.binary.v2+json
-You set the embedded data format when creating a consumer using the `consumers/_groupid_` endpoint--for more information, see the next section.
+You set the embedded data format when creating a consumer using the `consumers/groupid` endpoint--for more information, see the next section.
+
+The `Content-Type` must not be set if the `POST` request has an empty body. An empty body can be used to create a consumer with the default values.
```
Where can I find out about the default values?

ppatierno replied:
That's a good point @PaulRMellor .
It should be already mentioned somewhere else, otherwise we need it :-)
If the "name" is not specified, then a random one is generated by the bridge.
If the "format" is not specified, the "binary" is used.
For the other configurable Kafka consumer parameters, the defaults from Kafka official documentation are applied.

laidan6000 replied:
Good point. In my view, we should document the default values for Kafka bridge consumers in the API documentation at https://strimzi.io/docs/bridge/latest/#_createconsumer

I've created a separate issue for this.

As for the quickstart, I will briefly document the defaults in the Creating a Kafka Bridge consumer procedure.

laidan6000 replied:
Will document the default embedded data format in the Requests to the Kafka Bridge concept module.

PaulRMellor commented on books/con-requests-kafka-bridge.adoc:
```diff
@@ -48,13 +48,15 @@ m¦Content-Type: application/vnd.kafka.binary.v2+json
 The embedded data format is the format of the Kafka messages that are transmitted, over HTTP, from a producer to a consumer using the Kafka Bridge. Two embedded data formats are supported: JSON and binary.
 
-When creating a consumer using the `/consumers/_groupid_` endpoint, the `POST` request body must specify an embedded data format of either JSON or binary. This is specified in the `format` field, for example:
+When creating a consumer using the `/consumers/groupid` endpoint, the `POST` request body must specify an embedded data format of either JSON or binary. This is specified in the `format` field, for example:
```
We use `json' and `binary' in the equivalent producer description above.

PaulRMellor commented on books/con-requests-kafka-bridge.adoc:
```diff
@@ -87,14 +89,14 @@ Producer requests must also provide a `Content-Type` header that corresponds to
 ----
 
-Where the `embedded-data-format` is the embedded data format of the consumer: either `json` or `binary`.
+The `embedded-data-format` is either `json` or `binary`.
```
```suggestion
The __embedded data format__ is either `json` or `binary`.
```

PaulRMellor commented on books/proc-bridge-deleting-consumer.adoc:
```diff
@@ -0,0 +1,23 @@
+[id='proc-bridge-deleting-consumer-{context}']
+= Deleting a Kafka Bridge consumer
+
+Finally, delete the Kafa Bridge consumer that you used throughout this quickstart.
```
We don't have to though, do we?

ppatierno replied:
Why not? It's always better deleting a consumer when it's no longer used. It avoids having "zombies" on the bridge (even if there is a periodic check that remove these zombies not used for long time).

laidan6000 replied:
I'll leave as-is.

PaulRMellor commented on books/proc-configuring-kafka-bridge.adoc:
```diff
@@ -28,12 +26,11 @@ Use the properties file to specify Kafka and HTTP-related properties.
----
 Use:
++
 * `kafka.bootstrap.servers` to define the host/port connections to the Kafka cluster
```
Maybe we should be more specific about configuring these properties in another release?

PaulRMellor commented on books/proc-creating-kafka-bridge-consumer.adoc:
```diff
@@ -0,0 +1,53 @@
+----
++
+* The consumer is named `bridge-quickstart-consumer` and the embedded data format is set as `json`.
+* Some basic configuration settings are defined.
```
I wonder if this bullet is needed? We often use callouts to describe properties/parameters.

laidan6000 replied:
Agreed.

PaulRMellor commented on books/proc-deploying-kafka-bridge-quickstart.adoc:
```diff
@@ -0,0 +1,34 @@
+
+. Edit the `application.properties` file provided with the {ProductName} Kafka Bridge installation archive.
+
+. In the `HTTP related settings` use the default configuration:
```
It's not already set up? If so, maybe we say check the default settings?

PaulRMellor commented on books/proc-producing-messages-from-bridge-topics-partitions.adoc:
```diff
@@ -0,0 +1,94 @@
+}'
+----
++
+* `sales-lead-0001` is sent to a partition based on the hash of the key.
```
Not sure how `my-key` works in this instance. Where do we derive this value from?

ppatierno replied:
The producer can set a key with any value or anyway accordingly with the meaning of the key in the business application where the producer live (i.e. user-id, device-id, ...)

laidan6000 commented on books/proc-bridge-committing-consumer-offsets-to-log.adoc:
```diff
@@ -0,0 +1,29 @@
++
+Because no request body is submitted, offsets are committed for all the records that have been received by the consumer. Alternatively, the request body can contain an array (link:https://strimzi.io/docs/bridge/latest/#_offsetcommitseeklist[OffsetCommitSeekList^]) that specifies the topics and partitions that you want to commit offsets for.
++
+If the request is successful, the Kafka Bridge returns a `204` (No Content) code only.
```
Good suggestion, I'll update to the literal throughout.

laidan6000 commented:
Thanks for the review, @PaulRMellor. Apart from the unresolved conversations, I've made all the changes you suggested. 

@ppatierno - Please can you approve the PR?

ppatierno commented:
@laidan6000 there are some conflicts to fix before approving.

laidan6000 commented:
OK, I resolved a merge conflict in the `con-overview-kafka-brodge.adoc` file. This file was recently modified by @PaulRMellor, but in my branch I deleted it and added the contents to a new assembly named `assembly-kafka-bridge-overview`.

Looking in the changed files list, I see there are now three `generator/.` files. I think these were created by an extension in VS Code.

ppatierno commented:
@laidan6000 better avoiding them, please update the .gitignore to avoid this file are pushed and remove them from the PR.
